src/NServiceBus.FileBasedRouting/XmlRoutingFileParser.cs

```diff
@@ -42,23 +45,32 @@ public IEnumerable<EndpointRoutingConfiguration> Parse(XDocument document)
             return configs;
         }
 
-        static IEnumerable<Type> GetAllMessageTypes(XElement endpointElement, string singular, string plural)
+        IEnumerable<Type> GetAllMessageTypes(XElement endpointElement, string singular, string plural)
         {
             var handles = endpointElement.Element("handles");
 
             var separatelyConfiguredCommands = handles
                 ?.Elements(singular)
                 .Select(e => SelectCommand(e.Attribute("type").Value))
+                .Where(e => e != null)
```

Comment:
@SzymonPobiega hm that would work but I think an additional declaration adds quite a bit of pain to configure routing :( In that case I think I'd prefer going with @HEskandari suggested approach and log warnings.

@HEskandari I'm not aware of any validation regarding endpoint name. That's a good point. Although I think there's a bigger chance that you mispell some of the types and at least there we could help you by identifying that this type isn't around, it's a perfectly valid point that the assembly of a completely unrelated endpoint might not be around. Also, in case of commands, it should be less of an issue as send will also throw in case no route was found. So let's go ahead with your proposed changes 👍